File: Archiver/frmListenerTest.cs
using global::System;
using global::System.IO;
using global::System.Windows.Forms;
using Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public partial class FrmListenerTest
    {
        public FrmListenerTest()
        {
            InitializeComponent();
            if (File.Exists(@"C:\_ChangeLogs\FileChanges.dat"))
            {
                File.Delete(@"C:\_ChangeLogs\FileChanges.dat");
            }

            _Button1.Name = "Button1";
            _Button2.Name = "Button2";
            _btnViewLog.Name = "btnViewLog";
        }

        private FileSystemWatcher watchfolder = new FileSystemWatcher();

        private void Button1_Click(object sender, EventArgs e)
        {
            var dlg = new FolderBrowserDialog();
            if (dlg.ShowDialog() == DialogResult.OK)
            {
                txtDir.Text = dlg.SelectedPath;
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            StartLIstener2();
        }

        public void StartLIstener2()
        {

            // Dim watchfolder As = New System.IO.FileSystemWatcher()

            // this is the path we want to monitor
            watchfolder.Path = txtDir.Text;

            // Add a list of Filter we want to specify
            // make sure you use OR for each Filter as we need to
            // all of those 

            // watchfolder.NotifyFilter = IO.NotifyFilters.DirectoryName
            watchfolder.NotifyFilter = watchfolder.NotifyFilter | NotifyFilters.FileName;
            watchfolder.NotifyFilter = watchfolder.NotifyFilter | NotifyFilters.Attributes;

            // add the handler to each event
            watchfolder.Changed += logchange;
            watchfolder.Created += logchange;
            watchfolder.Deleted += logchange;

            // add the rename handler as the signature is different
            watchfolder.Renamed += logrename;

            // Set this property to true to start watching
            watchfolder.EnableRaisingEvents = true;
            Button2.Enabled = false;
            ckStop.Enabled = true;

            // End of code for btn_start_click
        }

        public void StopListener()
        {
            if (ckStop.Checked.Equals(true))
            {
                watchfolder.EnableRaisingEvents = false;
                Button2.Enabled = true;
                ckStop.Enabled = false;
            }
        }

        private void logchange(object source, FileSystemEventArgs e)
        {
            string Msg = "";
            if (e.ChangeType == WatcherChangeTypes.Changed)
            {
                Msg = "U|" + e.FullPath + "|";
                SaveChanges(Msg);
            }

            if (e.ChangeType == WatcherChangeTypes.Created)
            {
                Msg = "C|" + e.FullPath + "|";
                SaveChanges(Msg);
            }

            if (e.ChangeType == WatcherChangeTypes.Deleted)
            {
                Msg = "D|" + e.FullPath + "|";
                SaveChanges(Msg);
            }
        }

        public void logrename(object source, RenamedEventArgs e)
        {
            string Msg = "";
            Msg = "R|" + e.OldName + "|" + e.Name;
            SaveChanges(Msg);
        }

        private void SaveChanges(string Msg)
        {
            string TS = DateAndTime.Now.Ticks.ToString();
            string T = DateAndTime.Now.ToLongDateString().ToString();
            Msg = T + "|" + TS + "|" + Msg;
            if (!Directory.Exists(@"C:\_ChangeLogs"))
            {
                Directory.CreateDirectory(@"C:\_ChangeLogs");
            }

            string FQN = @"C:\_ChangeLogs\FileChanges.dat";
            try
            {
                using (var SW = new StreamWriter(FQN, true))
                {
                    SW.WriteLine(Msg);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Could not store data into file " + FQN + Constants.vbCrLf + ex.Message + ". Program terminating.", "Error", MessageBoxButtons.OK);
            }
        }

        private void btnViewLog_Click(object sender, EventArgs e)
        {
            string FQN = @"C:\_ChangeLogs\FileChanges.dat";
            try
            {
                lbOutput.Items.Clear();
                var reader = My.MyProject.Computer.FileSystem.OpenTextFileReader(FQN);
                string a;
                do
                {
                    a = reader.ReadLine();
                    lbOutput.Items.Add(a);
                }
                while (!(a is null));
                reader.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show("Could not store data into file " + FQN + ". Program terminating.", "Error", MessageBoxButtons.OK);
            }
        }
    }
}

File: Archiver/frmListenerTest.vb
Imports System.Windows.Forms
Imports System
Imports System.IO
Imports System.Diagnostics

Public Class FrmListenerTest

    Dim watchfolder As New System.IO.FileSystemWatcher()

    Sub New()
        InitializeComponent()

        If File.Exists("C:\_ChangeLogs\FileChanges.dat") Then
            File.Delete("C:\_ChangeLogs\FileChanges.dat")
        End If

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim dlg As New FolderBrowserDialog
        If (dlg.ShowDialog() = DialogResult.OK) Then
            txtDir.Text = dlg.SelectedPath
        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        StartLIstener2()
    End Sub

    Public Sub StartLIstener2()

        'Dim watchfolder As = New System.IO.FileSystemWatcher()

        'this is the path we want to monitor
        watchfolder.Path = txtdir.Text

        'Add a list of Filter we want to specify
        'make sure you use OR for each Filter as we need to
        'all of those 

        'watchfolder.NotifyFilter = IO.NotifyFilters.DirectoryName
        watchfolder.NotifyFilter = watchfolder.NotifyFilter Or IO.NotifyFilters.FileName
        watchfolder.NotifyFilter = watchfolder.NotifyFilter Or IO.NotifyFilters.Attributes

        ' add the handler to each event
        AddHandler watchfolder.Changed, AddressOf logchange
        AddHandler watchfolder.Created, AddressOf logchange
        AddHandler watchfolder.Deleted, AddressOf logchange

        ' add the rename handler as the signature is different
        AddHandler watchfolder.Renamed, AddressOf logrename

        'Set this property to true to start watching
        watchfolder.EnableRaisingEvents = True

        Button2.Enabled = False
        ckStop.Enabled = True

        'End of code for btn_start_click
    End Sub

    Sub StopListener()
        If ckStop.Checked.Equals(True) Then
            watchfolder.EnableRaisingEvents = False
            Button2.Enabled = True
            ckStop.Enabled = False
        End If
    End Sub

    Private Sub logchange(ByVal source As Object, ByVal e As System.IO.FileSystemEventArgs)

        Dim Msg As String = ""
        If e.ChangeType = IO.WatcherChangeTypes.Changed Then
            Msg = "U|" & e.FullPath & "|"
            SaveChanges(Msg)
        End If
        If e.ChangeType = IO.WatcherChangeTypes.Created Then
            Msg = "C|" & e.FullPath & "|"
            SaveChanges(Msg)
        End If
        If e.ChangeType = IO.WatcherChangeTypes.Deleted Then
            Msg = "D|" & e.FullPath & "|"
            SaveChanges(Msg)
        End If

    End Sub
    Public Sub logrename(ByVal source As Object, ByVal e As System.IO.RenamedEventArgs)
        Dim Msg As String = ""
        Msg = "R|" & e.OldName & "|" & e.Name
        SaveChanges(Msg)
    End Sub

    Private Sub SaveChanges(Msg As String)

        Dim TS As String = Now.Ticks.ToString
        Dim T As String = Now.ToLongDateString.ToString
        Msg = T + "|" + TS + "|" + Msg

        If Not Directory.Exists("C:\_ChangeLogs") Then
            Directory.CreateDirectory("C:\_ChangeLogs")
        End If

        Dim FQN As String = "C:\_ChangeLogs\FileChanges.dat"
        Try

            Using SW As New IO.StreamWriter(FQN, True)
                SW.WriteLine(Msg)
            End Using

        Catch ex As Exception
            MessageBox.Show("Could not store data into file " & FQN & Environment.NewLine + ex.Message & ". Program terminating.", "Error", MessageBoxButtons.OK)
        End Try

    End Sub

    Private Sub btnViewLog_Click(sender As Object, e As EventArgs) Handles btnViewLog.Click

        Dim FQN As String = "C:\_ChangeLogs\FileChanges.dat"
        Try
            lbOutput.Items.Clear()
            Dim reader As StreamReader = My.Computer.FileSystem.OpenTextFileReader(FQN)
            Dim a As String
            Do
                a = reader.ReadLine
                lbOutput.Items.Add(a)
            Loop Until a Is Nothing
            reader.Close()
        Catch ex As Exception
            MessageBox.Show("Could not store data into file " & FQN &
                      ". Program terminating.", "Error", MessageBoxButtons.OK)
        End Try

    End Sub
End Class